tests/integration/test_akafka_testutils.py

```diff
@@ -78,6 +90,150 @@ def test_delete_topics_all(kafka_fixture: KafkaFixture):  # noqa: F811
+
+    await kafka_fixture.publish_event(
+        payload=make_payload("clear1"), type_=TEST_TYPE, topic=topic_to_clear
+    )
+    await kafka_fixture.publish_event(
+        payload=make_payload("keep1"), type_=TEST_TYPE, topic=topic_to_keep
+    )
+
+    consumer = KafkaConsumer(
+        topic_to_keep,
+        topic_to_clear,
+        bootstrap_servers=kafka_fixture.kafka_servers[0],
+        group_id="test",
+        auto_offset_reset="earliest",
+        enable_auto_commit=True,
+    )
+
+    # Verify that both messages were consumed as expected so we can trust the consumer
+    count = 0
+    for record in consumer:
+        count += 1
+        assert record.topic in (topic_to_keep, topic_to_clear)
+
+        # break when we get to the end, not when we reach an expected record count
+        if consumer.position(partition_keep) == consumer.highwater(
+            partition_keep
+        ) and consumer.position(partition_clear) == consumer.highwater(partition_clear):
+            break
+    assert count == 2
+
+    # Publish new messages, one to CLEAR and one to KEEP
+    clear_payload2 = make_payload(topic_to_clear + "2")
+    await kafka_fixture.publish_event(
+        payload=clear_payload2, type_=TEST_TYPE, topic=topic_to_clear
+    )
+
+    keep_payload2 = make_payload(topic_to_keep + "2")
+    await kafka_fixture.publish_event(
+        payload=keep_payload2, type_=TEST_TYPE, topic=topic_to_keep
+    )
+
+    # Clear the CLEAR topic
+    kafka_fixture.clear_topics(topics=topic_to_clear)
+
+    # Set a reasonable timeout to check all messages but not run indefinitely
+    consumer.config["consumer_timeout_ms"] = 2000
+
+    # make sure the KEEP topic still has its event but CLEAR is empty
+    total = 0
+    for record in consumer:
+        total += 1
+        assert record.topic == topic_to_keep
+        assert record.value.decode("utf-8") == json.dumps(keep_payload2)
+
+    # Assert that we processed 1 message
+    assert total == 1
+    assert consumer.position(partition_clear) == consumer.highwater(partition_clear)
+    assert consumer.position(partition_keep) == consumer.highwater(partition_keep)
+
+    clear_payload3 = make_payload(topic_to_clear + "3")
+    await kafka_fixture.publish_event(
+        payload=clear_payload3, type_=TEST_TYPE, topic=topic_to_clear
+    )
+
+    keep_payload3 = make_payload(topic_to_keep + "3")
+    await kafka_fixture.publish_event(
+        payload=keep_payload3, type_=TEST_TYPE, topic=topic_to_keep
+    )
+
+    total = 0
+    keep_count = 0
+    clear_count = 0
+    for record in consumer:
+        total += 1
+        if record.topic == topic_to_keep:
+            keep_count += 1
+            assert record.value.decode("utf-8") == json.dumps(keep_payload3)
+        elif record.topic == topic_to_clear:
+            clear_count += 1
+            assert record.value.decode("utf-8") == json.dumps(clear_payload3)
+
+    assert total == 2
+    assert keep_count == 1
+    assert clear_count == 1
```

Comment:
See above, if you like to avoid working with counts.

Reply:
Check here: [bd4a52c](https://github.com/ghga-de/hexkit/pull/86/commits/bd4a52c0bc7b2b4ef76fafe5b390160f07205f92)